qt/widgets/instrumentview/src/InstrumentWidgetMaskTab.cpp

```diff
@@ -487,10 +510,131 @@ void InstrumentWidgetMaskTab::setActivity() {
     m_instrWidget->getSurface()->setInteractionMode(
         ProjectionSurface::DrawFreeMode);
     m_activeTool->setText("Tool: Free draw. " + whatIsBeingSelected);
+  } else if (m_pixel->isChecked()) {
+    m_activity = Pixel;
+    m_instrWidget->getSurface()->setInteractionMode(
+        ProjectionSurface::PickSingleMode);
+    m_activeTool->setText("Tool: Pixel mask. " + whatIsBeingSelected);
+  } else if (m_tube->isChecked()) {
+    m_activity = Tube;
+    m_instrWidget->getSurface()->setInteractionMode(
+        ProjectionSurface::PickTubeMode);
+    m_activeTool->setText("Tool: Tube/bank mask. " + whatIsBeingSelected);
   }
   m_instrWidget->updateInfoText();
 }
 
+/**
+ * Slot responding on the pick of a single detector pixel for masking.
+ * Can be used either to mask the picked pixel or its parent.
+ */
+void InstrumentWidgetMaskTab::singlePixelPicked(size_t pickID) {
+  auto &actor = m_instrWidget->getInstrumentActor();
+  const auto &componentInfo = actor.componentInfo();
+  if (!componentInfo.isDetector(pickID)) {
+    return;
+  }
+
+  QApplication::setOverrideCursor(QCursor(Qt::WaitCursor));
+  std::vector<size_t> detectorId{pickID};
+  m_instrWidget->updateInstrumentView(); // to refresh the pick image
+  auto wsMask = actor.getMaskWorkspace();
```

Comment:
For the local variables, whatever is const, it is a good practice to declare it explicitly. This conveys immediately that you are not going to modify it.